tests/test_constraints_discrete.py

```diff
@@ -238,3 +245,39 @@ def test_custom(campaign):
         & campaign.searchspace.discrete.exp_rep["Solvent_1"].eq("C3")
     ).sum()
     assert num_entries == 0
+
+
+@pytest.mark.parametrize(
+    "parameter_names",
+    [
+        [
+            "Temperature",
+            "Solvent_1",
+            "Solvent_2",
+            "Solvent_3",
+            "Fraction_1",
+            "Fraction_2",
+            "Fraction_3",
+        ]
+    ],
+)
+@pytest.mark.parametrize(
+    "constraint_names",
+    [
+        ["Constraint_4"],
+        ["Constraint_12"],
+        ["Constraint_4", "Constraint_8", "Constraint_9"],
+    ],
+)
+def test_polars_product(constraints, parameters):
+    ldf = parameter_cartesian_prod_polars(parameters)
+    pol_df = ldf.collect()
+    pand_df = parameter_cartesian_prod_to_df(parameters)
+
+    assert len(pol_df) == len(pand_df)
+
+    _apply_pandas_constraint_filter(pand_df, constraints)
+
+    pol_df = _apply_polars_constraint_filter(ldf, constraints)
+
+    assert len(pol_df) == len(pand_df)
```

Comment:
strinclty this test shoukd test via merge that the dataframes are fully equal, this just tests length

Reply:
I used Pandas' `assert_frame_equal()` that is more efficient than merging.